File: Chamomile.Common/PromptOrderData.cs
namespace Chamomile.Common {

    /// <summary>Small trailer hitch to add information on where this prompt was copied from</summary>
    public class PromptOrderData {

        /// <summary>
        /// Source of this Prompt: "IMAGE_BASE","IMAGE","SAVED_PROMPT","PROMPTBOX"
        /// </summary>
        public string Source { get; set; } = "";

        /// <summary>Image this prompt was based on, or sample image of the saved prompt. -1 if from the promptbox </summary>
        public int Sample { get; set; } = -1;

        /// <summary>Albums to add this image to after generation. Usually will either be of length 0 or 1</summary>
        public List<int>? Albums { get; set; } = [];
    }
}

File: chamomile.web/src/model/PromptOrderData.ts
export type PromptSource = "IMAGE_BASE" | "IMAGE" | "SAVED_PROMPT" | "PROMPTBOX" | "UNKNOWN" | "UPLOAD" | "GRID";

export default interface PromptOrderData {
    source: PromptSource
    sample: number
    albums: number[]
    gridId?: number;
    xPos?: number;
    yPos?: number;
    xVal?: string;
    yVal?: string;
    model?: string
}